jgrapht-core/src/main/java/org/jgrapht/alg/color/SaturationDegreeColoring.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * (C) Copyright 2017-2017, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.color;
+
+import java.lang.reflect.Array;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.NavigableSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+import org.jgrapht.alg.interfaces.VertexColoringAlgorithm;
+
+/**
+ * The Dsatur greedy coloring algorithm.
+ * 
+ * <p>
+ * This is the greedy coloring algorithm using saturation degree ordering. The saturation degree of
+ * a vertex is defined as the number of different colors to which it is adjacent. The algorithm
+ * selects always the vertex with the largest saturation degree. If multiple vertices have the same
+ * maximum saturation degree, a vertex of maximum degree in the uncolored subgraph is selected.
```

Comment:
I changed the implementation to be a little bit more efficient. I also documented that this is O(n^2). 

There is a slightly faster implementation (for sparse graphs) which is O(m log n) but requires to implement a customized binary search tree with intervals. I added a reference to the paper, maybe we can do that in the future.